src/Microsoft.AspNet.Http/Features/QueryFeature.cs

```diff
@@ -41,11 +41,23 @@ void IFeatureCache.CheckFeaturesRevision()
         {
             if (_cachedFeaturesRevision != _features.Revision)
             {
-                _request = null;
-                _cachedFeaturesRevision = _features.Revision;
+                ResetFeatures();
             }
         }
 
+        public void UpdateFeatures(IFeatureCollection features)
```

Comment:
Ah, gotcha. You still have two classes of objects though, IFeatureCache { CheckFeaturesRevision } and IUpdatableFeatureCache : IFeatureCache { UpdateFeatures }. Only the managers will be updated.